### PR TITLE
ENH add pip check and lower fitsio requirement

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
Seeing if this works.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,12 +31,16 @@ requirements:
     - fitsio >=1.1.2
     - matplotlib-base >=3.3.0
     - lsstdesc.coord >=1.0.5
-    - galsim >=2.3
+    - galsim >=2
```
Comment (beckermr) [inline recipe/meta.yaml]:
```suggestion
    - galsim =2.2
```

**Comment by beckermr** [comment]
doesn't work but doesn't harm the build

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
cc @rmjarvis

**Comment by rmjarvis** [comment]
Hm.  I can't get `pip check` to report that dependencies are not up to date.  I intentionally downgraded to invalid versions and it still just outputs
```
$ pip check
No broken requirements found.
```
Maybe it's not checking versions, just whether things are installed or not?  If I completely uninstall matplotlib, I get
```
$ pip check
healpy 1.15.0 requires matplotlib, which is not installed.
piff 1.1.0 requires matplotlib, which is not installed.
```
But if 3.2 is installed, it thinks that's ok.

**Comment by beckermr** [comment]
`pip check` relies on metadata in the pip-generated files upon installation. It doesn't know about conda. You have to set the version constraints here: https://github.com/rmjarvis/Piff/blob/releases/1.1/setup.py#L286

**Comment by beckermr** [comment]
Be aware that `pip check` can generate false positives due to it not recognizing some conda-installed stuff. Also, it checks the entire environment, not just the package in question. Thus if one of your deps doesn't pass, it will fail.

**Comment by rmjarvis** [comment]
I thought pip used requirements.txt, not the setup.py line.  (I thought the latter was for `setup.py install`.)  Has that changed recently?  Or was I just wrong?  Is requirements.txt obsolete now?

**Comment by beckermr** [comment]
I don't think pip ever used that but I could be mistaken.

**Comment by rmjarvis** [comment]
https://www.idkrtm.com/what-is-the-python-requirements-txt/

But :shrug:.  I don't actually know anything on this front.  I always feel like I'm treading water on these issues.  Anyway, this all seems fine.  I'll update setup.py to have the required versions right, so starting with the next release, it will start being relevant.

**Review by rmjarvis — APPROVED** [review]

**Comment by beckermr** [comment]
Right. I don't think requirements.txt is anything special here that setuptools or pip understands to use to effect the build. I think it is just convention that everyone calls that file that. This is unlike say the MANIFEST.in file which has direct effects on what a given build of a package has in it.

**Pull request merged**